kdwsdl2cpp/common/fileprovider.cpp

```diff
@@ -94,30 +97,38 @@ bool FileProvider::get( const QUrl &url, QString &target )
     mFileName = target;
   }
 
-  qDebug("Downloading '%s'", url.toEncoded().constData());
-
-  QNetworkAccessManager manager;
-  QNetworkRequest request(url);
-  QNetworkReply* job = manager.get(request);
-
-  QEventLoop loop;
-  connect(job, SIGNAL(finished()),
-          &loop, SLOT(quit()));
-  loop.exec();
-
-  if (job->error()) {
-      qWarning("Error downloading '%s': %s", url.toEncoded().constData(), qPrintable(job->errorString()));
-      return false;
+  QByteArray data;
+  if (fileProviderCache.contains(url)) {
```

Comment:
Nitpick: this is a double lookup, better written as
```
QByteArray data = fileProviderCache.value(url);
if (!data.isEmpty()) {
    qDebug...
} else {
```

Reply:
That makes an assumption about valid data. Theoretically, the downloaded data could be empty. Using the QMap::find and iterators is the proper way to do this.

Reply:
You're right, this is a case where the map _could_ contain an empty value. Would likely give a parse error down the line though ;)
OK, let's leave it at that.